go/integration/pkcs11kms/key.go

```diff
@@ -0,0 +1,34 @@
+package pkcs11kms
+
+import (
+	"github.com/google/uuid"
+	"net/url"
+)
+
+// Key used to map the location of the key in the HSM, and optionally provide it's protected wrapped key
+type Key struct {
+	id          uuid.UUID
+	wrappedBlob []byte
+}
+
+// key URI for pkcs11 is pcks11://keyid pkcs11://keyid?blob=<jose.JWK.String()>
+
+func parseKeyURI(keyURI string) (k *Key, err error) {
+	k = &Key{}
+	var u *url.URL
+	if u, err = url.Parse(keyURI); err != nil {
+		return
+	}
+	if k.id, err = uuid.Parse(u.Host); err != nil {
+		return
+	}
+
+	if q := u.Query(); q != nil {
```

Comment:
if `q` is nil, then the function returns `nil, nil` 

this is not a good Go convention as one would expect either a non nil error or a non nil *Key as a return value

Reply:
I'm not sure I follow. 

If `q` is nil, then the function will still return `*Key,nil` as k is  defined on line 17, nil as there is no "wrappedKey" for the wrapped key mode.

The expected outcome is a *Key with optional wrappedKey(from a query param), or an error.

Reply:
I see, I should've read more closely the comment of the function saying: `optionally provide it's protected wrapped key`.

looks good then.

Reply:
no worries, just wanted to make sure i wasn't missing something :)

appreciate the review.